pkg/oidc/check/flow.go

```diff
@@ -0,0 +1,161 @@
+package check
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/pkg/browser"
+	"github.com/weaveworks/weave-gitops/pkg/logger"
+	"github.com/weaveworks/weave-gitops/pkg/server/auth"
+	"golang.org/x/oauth2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Options struct {
+	ClientID                   string
+	ClientSecret               string
+	IssuerURL                  string
+	SecretName                 string
+	SecretNamespace            string
+	Scopes                     []string
+	ClaimUsername              string
+	ClaimGroups                string
+	OpenURL                    func(string) error
+	InsecureSkipSignatureCheck bool
+}
+
+type Claims struct {
+	Username string
+	Groups   []string
+}
+
+// GetPrincipal retrieves OIDC claims by sending the user through an authorization code flow. It spins
+// up a temporary web server, sets the server's address as redirect URI in the authentication request
+// and subsequently exchanges the authorization code for an ID token.
+// NOTE: Make sure to configure your OIDC provider so that it accepts "http://localhost:9876" as redirect URI.
```

Comment:
I'm not sure this will be shown to the user?

Reply:
Oh, I thought when you asked for this to be added you wanted me to add it to the code comment. Added it to the long usage text.

Reply:
The user creating the OIDC client/secret usually needs to configure the allowed redirectURLs in the OIDC provider?

Otherwise they _usually_ get an error, give it a try :-)